Sources/Network/Http/UrlSessionHttp.swift

```diff
@@ -164,10 +167,23 @@ open class UrlSessionHttp: Http {
     // MARK: - Task
 
     private class Task: HttpTask {
+        var uploadBytes: Int64 = -1
+        var uploadTotalBytes: Int64 = -1
+        var uploadProgress: HttpProgress?
```

Comment:
Confusing name. I'd suggest `uploadProgressCallback` or `uploadProgressChanged`

Reply:
@bealex How about this
```swift
struct Progress {
    var bytes: Int64
    var totalBytes: Int64
    var callback: HttpProgress?
}

class Task {
    var uploadProgress: Progress
    var downloadProgress: Progress
    // ...
}
```

Reply:
```swift
struct Progress {
    var bytes: (Int64, total: Int64)?
    var callback: HttpProgress?
}
```
?

Reply:
Looks wrong. `bytes` could exist without `totalBytes`.

Reply:
Damn. Ok, maybe something like this then?
```swift
enum Progress {
    case unknown
    case finished
    case indeterminate(bytes: UInt64)
    case determinate(bytes: UInt64, total: UInt64)
}
```
It starts to look over engineered a little bit.

Reply:
Too complex 😉

Reply:
Yes, but at least noone will miss a case. :)